File: src/main/java/land/leets/domain/portfolio/usecase/GetPortfolios.java
package land.leets.domain.portfolio.usecase;

import land.leets.domain.portfolio.presentation.dto.PortfolioResponse;
import land.leets.domain.portfolio.presentation.dto.PortfoliosResponse;

import java.util.List;

public interface GetPortfolios {
    List<List<PortfoliosResponse>> all(String generation);

    PortfolioResponse one(Long portfolioId);
}

File: src/main/kotlin/land/leets/domain/portfolio/usecase/GetPortfolios.kt
package land.leets.domain.portfolio.usecase

import land.leets.domain.portfolio.presentation.dto.PortfolioResponse
import land.leets.domain.portfolio.presentation.dto.PortfoliosResponse

interface GetPortfolios {
    fun all(generation: String?): List<List<PortfoliosResponse>>

    fun one(portfolioId: Long): PortfolioResponse
}
